modules/apps/portal/portal-upload/src/main/java/com/liferay/portal/upload/internal/ServletFileUploadImpl.java

```diff
@@ -63,6 +70,14 @@ public List<FileItem> parseRequest(
 
 			return fileItems;
 		}
+		catch (FileUploadBase.SizeLimitExceededException
+					sizeLimitExceededException) {
+
+			throw new BadRequestException(
```

Comment:
This exception is a JAX RS exception (headless layer) which you moved from `MultipartBodyMessageBodyReader`. This exception was being caught there to provide a headless friendly message. I don't think that code belongs here. ServletFileUploadImpl is used in other situations where headless exceptions make no sense.

I would move the catch back to its original place so that headless decides how to handle it, while fixing also the other comment above.

Reply:
I've moved the catch back to its original place to preserve the headless friendly error message